authors/apps/profiles/views.py

```diff
@@ -68,3 +52,92 @@ class ListProfileView(generics.ListAPIView):
+
+    def put(self, request):
+        """Method to follow a user's profile
+        """
+        username = request.data.get('username', {})
+        print(username)
+
+        if request.user.username == username:
+            response_data = {
+                'message': 'Sorry, you cannot follow yourself!'}
+
+            return Response(data=response_data,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        toggler_response = Profile.objects.toggle_follow(
+            request.user, username)
+        if toggler_response is None or toggler_response == []:
+            response_data = {
+                'message': 'Please enter a valid username!'}
+
+            return Response(data=response_data,
+                            status=status.HTTP_400_BAD_REQUEST)
+        if not toggler_response[1]:
+            response_data = {
+                'message': 'You have unfollowed {}!'.format(username)}
+
+            return Response(data=response_data, status=status.HTTP_200_OK)
+        else:
+            response_data = {
+                'message': 'You have followed {}!'.format(username)}
+            return Response(data=response_data, status=status.HTTP_200_OK)
+
+
+class RetrieveFollowersView(generics.ListAPIView):
+    """View class for retrieving all followers
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        """Method to get and return all a user's followers
+        """
+        user = request.user
+
+        user_id = User.objects.filter(email=user.email).values('id')[
+            0]['id']
+        queryset = Profile.objects.get(user__pk=user_id).followers.all()
+
+        followers = []
+        for follower in queryset:
+            follower_name = User.objects.get(email=follower.email)
+            follower_username = follower_name.username
+            followers.append(follower_username)
+        if followers is None or followers == []:
+            msg = {'error': 'No followers were found!'}
+            return Response(data=msg, status=status.HTTP_400_BAD_REQUEST)
+        msg = {'followers': followers}
+        return Response(data=msg, status=status.HTTP_200_OK)
+
+
+class RetrieveFollowingView(generics.ListAPIView):
+    """View class for retrieving all followers
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        """Method to get and return all a user's followers
+        """
+        user = request.user
+
+        user_id = User.objects.filter(email=user.email).values('id')[
+            0]['id']
+        queryset = User.objects.filter(pk=user_id).first().is_following.all()
+
+        following = []
+        for follower in queryset:
+            following_name = User.objects.get(username=follower)
+            following_username = following_name.username
+            following.append(following_username)
+        if following is None or following == []:
+            msg = {'error': 'No one is following you yet!'}
```

Comment:
@BarnaTB Change the key here to message. It is not an error when no one is following you.
```suggestion
            msg = {'message': 'No one is following you yet!'}
```